versal_net.mk

```diff
@@ -0,0 +1,191 @@
+
+tfa:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) bl31
+
+tfa-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+################################################################################
+# OP-TEE
+#################################################################################
+
+optee-os: optee-os-common
+
+optee-os-clean: optee-os-clean-common
+
+################################################################################
+# U-Boot
+################################################################################
+
+U-BOOT_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+U-BOOT_DTS = xilinx-versal-net-virt
+U-BOOT_DEFCONFIG_COMMON_FILES = \
+	$(UBOOT_PATH)/configs/xilinx_versal_net_virt_defconfig \
+	$(CURDIR)/kconfigs/versal.conf
+
+$(UBOOT_PATH)/.config: $(U-BOOT_CONFIG_COMMON_FILES)
+	cd $(UBOOT_PATH) && \
+		$(U-BOOT_EXPORTS) \
+		scripts/kconfig/merge_config.sh $(U-BOOT_DEFCONFIG_COMMON_FILES)
+
+u-boot: u-boot-defconfig
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) DEVICE_TREE=$(U-BOOT_DTS) DTC_FLAGS="-@"
+
+u-boot-clean: u-boot-defconfig-clean
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) clean
+
+u-boot-defconfig: $(UBOOT_PATH)/.config
+
+.PHONY: u-boot-defconfig-clean
+u-boot-defconfig-clean:
+	rm -f $(UBOOT_PATH)/.config
+
+###############################################################################
+# Device-Tree
+###############################################################################
+dtbo: linux
+	${LINUX_PATH}/scripts/dtc/dtc -@ -I dts -O dtb -o versal/versal-optee.dtbo versal/versal-optee.dtso
+
+dtbo-clean:
+	rm -f versal/versal-optee.dtbo
+
+################################################################################
+# Linux kernel
+################################################################################
+
+LINUX_DEFCONFIG_COMMON_ARCH := arm64
+LINUX_DEFCONFIG_COMMON_FILES := \
+		$(LINUX_PATH)/arch/arm64/configs/xilinx_versal_net_defconfig \
+		$(CURDIR)/kconfigs/versal.conf
+
+linux-defconfig: $(LINUX_PATH)/.config
+
+LINUX_COMMON_FLAGS += ARCH=arm64 -j8
+
+linux: linux-common
+
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm64
+
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm64
+
+linux-cleaner: linux-cleaner-common
+
+###############################################################################
+# Buildroot
+###############################################################################
+
+BR2_TARGET_GENERIC_ISSUE ?= "OP-TEE embedded distrib for $(PLATFORM)"
+BR2_TARGET_ROOTFS_EXT2 ?= y
+BR2_PACKAGE_BUSYBOX_WATCHDOG ?= y
+
+# TF-A, Linux kernel, U-Boot and OP-TEE OS/Client/... are not built from their
+# related Buildroot native package.
+BR2_TARGET_ARM_TRUSTED_FIRMWARE ?= n
+BR2_LINUX_KERNEL ?= n
+BR2_TARGET_OPTEE_OS ?= n
+BR2_TARGET_UBOOT ?= n
+BR2_PACKAGE_OPTEE_CLIENT ?= n
+BR2_PACKAGE_OPTEE_TEST ?= n
+BR2_PACKAGE_OPTEE_EXAMPLES ?= n
+BR2_PACKAGE_OPTEE_BENCHMARK ?= n
```

Comment:
This one is deprecated and should be removed

Reply:
Ok, will do.